### ISSUE TITLE
Relate nullness of `map.get(k)` to the declaration `@KeyFor("map") k`

### ISSUE THREAD

**Issue by GoogleCodeExporter** [typetools/checker-framework#372]
```
What steps will reproduce the problem?
1. Run the Null Checker on the class below
import java.util.HashMap;
import java.util.Map;

import org.checkerframework.checker.nullness.qual.EnsuresNonNull;
import org.checkerframework.checker.nullness.qual.KeyFor;

class Test {
    private final Map<String, String> labels = new HashMap<>();

    @EnsuresNonNull("labels.get(#1)")
    void foo(@KeyFor("labels") String v) {
        labels.put(v, "");
    }
}

What is the expected output? What do you see instead?
Expected: No warning
Actual: Warning: the postcondition about 'this.labels.get(v)' of this method is 
not satisfied void foo(@KeyFor("labels") String v) {

What version of the product are you using? On what operating system?
1.8.6/Windows 8/Java 8


```

Original issue reported on code.google.com by `ClovisSe...@gmail.com` on 4 Oct 2014 at 10:48

**Comment by mernst** [comment]
I have added a test case as file `checker/tests/nullness/flow/MapGet.java`.
